test/compute/operation.js

```diff
@@ -406,5 +406,38 @@ describe('Operation', function() {
         operation.startPolling_();
       });
     });
+
+    describe('operation progress', function() {
+      var apiResponse = { progress: 50 };
+
+      beforeEach(function() {
+        delete operation.progress;
+
+        operation.getMetadata = function(callback) {
+          callback(null, apiResponse, apiResponse);
+        };
+      });
+
+      it('should emit progress with metadata', function(done) {
+        operation.on('progress', function(progress, metadata) {
+          assert.strictEqual(operation.progress, apiResponse.progress);
+          assert.strictEqual(progress, apiResponse.progress);
+          assert.strictEqual(metadata, apiResponse);
+          done();
+        });
+
+        operation.startPolling_();
+      });
+
+      it('should only emit when progress has updated', function(done) {
+        operation.on('progress', function() {
+          // if done is called multiple times the test will fail
+          done();
+        });
+
+        operation.startPolling_();
+        operation.startPolling_();
+      });
```

Comment:
While this is absolutely not as cool, I would rather see something more explicit:

```js
this.timeout(5000);

var polledCount = 0;
var timesToPoll = 4;
var progressesEmitted = [];

operation.on('progress', function(progress) {
  progressesEmitted.push(progress);
});

operation.getMetadata = function () {
  polledCount++;

  if (polledCount === timesToPoll) {
    callback(null, { state: 'DONE' });

    setImmediate(function() {
      assert.deepEqual(progressesEmitted, [33, 66]);
      done();
    });

    return;
  }

  if (polledCount < timesToPoll / 2) {
    callback(null, { progress: 33 });
  } else if {
    callback(null, { progress: 66 });
  }
};
```

(un-tested, hope it (or something like it) works!)